### PR TITLE
[lit-next] Renames `platform-support` to `polyfill-support`

### LINKED ISSUE #1526: [lit-next] Rename platform-support to polyfill-support
## Description
Rename `platform-support.js` files to `polyfill-support.js`, to be more explicit about their purpose.

## Acceptance criteria
Rename checked in; relevant docs updated.

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #1526.

### CODE REVIEW

File: packages/lit-element/CHANGELOG.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - [Breaking] Removed build support for TypeScript 3.4.
 - [Breaking] Decorators are no longer exported from the `lit-element` module. Instead, import any decorators you use from `lit-element/decorators/*`.
 - [Breaking] `lit-html` has been updated to 2.x.
-- [Breaking] Support for running in older browsers has been removed from the default configuration. Import the `platform-support` module to support Shady DOM. Note also that Lit parts inside `<style>` elements are no longer supported.
+- [Breaking] Support for running in older browsers has been removed from the default configuration. Import the `polyfill-support` module to support Shady DOM. Note also that Lit parts inside `<style>` elements are no longer supported.
```
Comment (justinfagnani) [inline packages/lit-element/CHANGELOG.md:58]: I'd add a new item and a (Since... ) note and we can collapse these at release

**Review by justinfagnani — APPROVED** [review]

File: packages/lit-element/CHANGELOG.md (hunk 1 above)
Comment (sorvell) [reply packages/lit-element/CHANGELOG.md:58]: Done.

**Review by kevinpschaaf — APPROVED** [review]
LGTM

**Comment by sorvell** [comment]
Note, merging even though Safari 13 tests are failing on Sauce Labs since they pass locally. This issue is being separately investigated.

**Pull request merged**